### DISCUSSION TITLE
setup_amps fails while attempting to bias hemt2 on sat1 crate 2 slot 2

### DISCUSSION THREAD

**Discussion by jennamo** [simonsobs/sodetlib#374] [Issues and Debugging]
snippet of log from attempt at running `setup_amps`
<img width="535" alt="Screenshot 2023-08-06 at 12 03 48 PM" src="https://github.com/simonsobs/sodetlib/assets/25230181/db9b48f4-9b4d-4c16-b4b5-eaabbec6f340">

manually setting voltages for both hemt amps results in pretty significantly different current draw (although maybe this is within expectations for this particular amplifier model?)
<img width="405" alt="Screenshot 2023-08-06 at 12 05 25 PM" src="https://github.com/simonsobs/sodetlib/assets/25230181/11e5101e-2658-4c37-afc1-5477b95e238f">

manually setting both hemt drain/gate voltages to 0:
<img width="508" alt="Screenshot 2023-08-06 at 12 08 56 PM" src="https://github.com/simonsobs/sodetlib/assets/25230181/3ab7644c-245f-4ab2-98f5-aabf4b19db31">

disabling all amps:
<img width="440" alt="Screenshot 2023-08-06 at 12 09 41 PM" src="https://github.com/simonsobs/sodetlib/assets/25230181/e7ab9eb8-f497-454c-ab67-86bfbfceeb3f">

**Comment by jlashner** [comment]
Ok, I tried setting voltages manually, and found that it seems like the drain current for hemt2 is completely independent of gate voltage. I was able to achieve a drain current of 8mA by lowering the drain current to ~.3 V, but this kind of seems like a faulty amplifier. 
![image](https://github.com/simonsobs/sodetlib/assets/4718487/481f439c-9246-4768-9b37-3c4c975385c1)
Jenna is attempting to reseat the bias cables and then we'll try again, but might not be able to read out this half-UFM for this cooldown.

**Comment by dpdutcher** [comment]
I just want to flag that I ran into a similar problem here at Princeton, but that it seems to be an issue with the RTM/SMuRF board because another board could power that amp just fine. The board didn't show that issue in the past but now does.

**Comment by jlashner** [comment]
Ok it turned out re-seating the DC bias cables seemed to fix the problem, so it could have just been a bad connection

**Comment by jennamo** [comment]
circling back here because we are now seeing the same issue on crate 1, all slots. reseating cables doesn't fix. going to try probing VD/VG output using cryocard breakout board.

**Comment by jennamo** [comment]
update, probing with cryocard breakout board registers no voltages on any of the gates, and small (<1V) on the drains, even when drains/gates are set to +/- V.

**Comment by jennamo** [comment]
we tried swapping SCSI cables and cryocards and continued seeing the same behavior. ultimately, we switched RTMs and are now able to bias amplifiers correctly.

**Reply by msilvafe** [reply]
Sorry catching up here. Was the conclusion that this was a faulty RTM and we swapped it with a spare and sent it back to slac for repair?

**Reply by jennamo** [reply]
We pulled the RTM and marked it as suspicious. I believe @swh76 is going to do/has done some diagnostics now that he is at site?

**Reply by jlashner** [reply]
Shawn did probe the voltages of all the RTMs, and they were all good except for the one that was consistently causing problematic voltages for us. @swh76 should comment with which RTM, it was, but a couple of the voltage rails were fluctuating wildly. He removed and inspected the RTM and found that some of the traces looked bad, and we replaced the bad RTM with a new one and he brought the bad one back to be inspected at SLAC.

We should probably note here the id of the bad one  and the replacement, along with which slot it was running in. I think they are written in shawn's notes.
